### PR TITLE
Log the goroutines stack when a healthprobe fails

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

When a component is unhealthy, log the goroutines stacktrace so that we can understand where it is stuck.

### Motivation

With #7605, the only reason for a component to be unhealthy should be bug (the goroutine exited abnormally or is definitively blocked somewhere). In such a situation, we need the stacktrace to understand where the goroutine is stuck.

### Additional Notes

Ideally, I would have preferred to log only the stacktrace of the stuck goroutine but I only found functions to dump the stacktrace of the current goroutine or all the goroutines.

This change is the piece of code that I used to get the stacktraces in the description of #7605.

This log is extremely verbose but we have evidences that unhealthy components is quite uncommon.
Indeed, on Kubernetes, components health is used as a liveness probe so that, when a component is unhealthy, Kubernetes kills the agent and restart it.
If it was happening often, it would have already been a problem.

### Describe your test plan

This PR can be tested only without #7605. In this case, following the testing instructions of it would lead some components to become unhealthy and the stacktraces should be logged if and only if `log_all_goroutines_when_unhealthy` is set to `true`.

### CODE REVIEW

**Comment by L3n41c** [comment]
Given the amount of logs this feature can generate, let’s gate it behind a config flag.

**Review by sgnn7 — CHANGES_REQUESTED** [review]
@L3n41c  Tiny nit comment on the code but I fully agree that we have to have an opt-in flag otherwise we could easily flood the machine with logs even for an intermittent issue.

**Review by vboulineau — APPROVED** [review]

File: pkg/api/healthprobe/healthprobe.go [hunk 1]
```diff
@@ -70,6 +70,7 @@ func healthHandler(getStatusNonBlocking func() (health.Status, error), w http.Re
 	if len(health.Unhealthy) > 0 {
 		w.WriteHeader(http.StatusInternalServerError)
 		log.Infof("Healthcheck failed on: %v", health.Unhealthy)
+		log.Infof("goroutines stack: \n%s\n", allStack())
```
[the remark this answers is not in the document]
Comment (L3n41c) [reply pkg/api/healthprobe/healthprobe.go]: Fixed!

**Comment by L3n41c** [comment]
I’ve now gated the new log behind a config parameter to avoid flooding the logs: https://github.com/DataDog/datadog-agent/pull/7825/commits/97bc19d4b17753b16a17fbab19be348eea3b9d51

**Review by sgnn7 — APPROVED** [review]

**Review by apigirl — APPROVED** [review]
👍 from docs

**Pull request merged**